### PR TITLE
Add ocamlformat formatter

### LINKED ISSUE #204: Formatter for OCaml
**Formatter tool**
[ocamlformat](https://github.com/ocaml-ppx/ocamlformat)

**Filetype(s)**
OCaml

**Additional context**
`ocamlformat` does not seem to support range formatting. It reads formatting settings from `.ocamlformat` file, which it searches for in input file directory, going up into parent directores until it finds one. When formatting stdin, one of `--name`, `--impl`, or `--intf` options has to be provided.

### PR DESCRIPTION
**Pull request by dimitrijer**
Fixes #204.

### CODE REVIEW

File: autoload/codefmt/ocamlformat.vim [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    let l:executable = s:plugin.Flag('ocamlformat_executable')
+
+    " ocamlformat requires --name, --impl, or --intf when reading from
+    " stdin.
+    let l:inputflags = ['--name', @%]
+    if len(@%) == 0
+      " Assume we're formatting an implementation file.
+      let l:inputflags = ['--impl']
+    endif
+    call codefmt#formatterhelpers#Format(
```
Comment (dbarnett) [inline autoload/codefmt/ocamlformat.vim:53]: If this hits some error in formatting, it'd blow up with an ugly error. For now could you add a TODO to catch and handle errors more nicely?

**Review by dbarnett — CHANGES_REQUESTED** [review]
One question about error handling. Looks good otherwise.

File: autoload/codefmt/ocamlformat.vim (hunk 1 above)
Comment (dimitrijer) [reply autoload/codefmt/ocamlformat.vim:56]: Good catch, done.

**Comment by dbarnett** [comment]
Thanks!

**Pull request merged**